src/mono/wasm/runtime/run.ts

```diff
@@ -48,20 +45,22 @@ export async function mono_run_main(main_assembly_name: string, args?: string[])
             args = [];
         }
     }
+    
+    // TODO MF: Fix mono_run_main
```

Comment:
@pavelsavara  Do you know of a way to "conditionaly import from different file in typescript"? Something like what we can accomplish in C# with partial classes + conditional file include in csproj?

One way to solve this in JS/TS world I can think of is by going to OOP. Declare interfaces/helper objects for thing we might need to swap in NAOT, but that is slightly out of the scope for this PR.

Reply:
you are talking about 2 different concepts, I think.
1) generating different code
2) taking another code path while both are available at the runtime. 

Re: 1)
I strongly prefer this option, because of the download size.
We have ways how to achieve that with RollupJS, I think.

```ts
import IsNativeAOT from "consts:NativeAOT";

export const myFunction = IsNativeAOT
    ? function myNaotFunction() { return null }
    :  function myMonoFunction() { return null }
```

Would this work ?

Ideally that should be in some top level file, but our TS is bit of spaghetti WRT file dependencies.
At the moment those are
- loader/index.ts
- exports.ts
- exports-linker.ts
- types/export-types.ts - you can ignore types I guess
- net6-legacy/export-types.ts - you can ignore types I guess

Re 2)
- let's not do classes in low level runtime JS/TS. Perhaps interfaces, but do we really need that instead of simple `if` condition in few places calling the other implementation ? How many methods you need to differentiate ?

Reply:
Obviously `myNaotFunction` could be in different file than `myMonoFunction`. The fact you include both files doesn't matter because tree shaking will remove it.

Reply:
I don't want to have them both on runtime. I was referring to a similar pattern you mentioned with the if, but in a way that only runtimelab branch can define the overrides. Something like that upstream have `api.myFunction = monoFunction` and here we can override it `api.myFunction = naotFunction`.